### PR TITLE
Add an unnecessary imports migration guide

### PR DESCRIPTION
**Pull request by srawlins**

### CODE REVIEW

File: src/docs/release/breaking-changes/unnecessary-imports.md [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+<!-- skip -->
+```dart
+import 'package:flutter/widgets.dart';
+
+void f(Characters characters, Actions actions) {}
+```
+
+## Timeline
+
+Landed in version: xxx<br>
```
Comment (srawlins) [inline src/docs/release/breaking-changes/unnecessary-imports.md:70]: I guess wait for unnecessary_imports to land before merging this?

Comment (sfshaza2) [reply src/docs/release/breaking-changes/unnecessary-imports.md:70]: Yup. We need a release version listed here.

Comment (domesticmouse) [reply src/docs/release/breaking-changes/unnecessary-imports.md:70]: Marking as draft until we have a release version of `unnecessary_imports `. 

@srawlins please mark this PR as ready to review when the release version lands. Thanks!

**Comment by sfshaza2** [comment]
Where are we on this PR, @srawlins?

**Comment by sfshaza2** [comment]
FYI. Sam, I am freezing the website repo next Friday (Aug 13th). We must either land this by then or you'll have to submit another PR. (We are changing the website infrastructure, so all remaining PRs will be landed or closed.)

**Comment by srawlins** [comment]
Definitely not this week. I'll close and re-open.

**Comment by sfshaza2** [comment]
FYI, you can't "re-open" because the docs are moving to a new repo. But you can save the content of your current PR and open on the new repo. Thx!!

**Pull request closed without merging**